<!DOCTYPE HTML>
<html lang="en" class="light sidebar-visible" dir="ltr">
    <head>
        <!-- Book generated using mdBook -->
        <meta charset="UTF-8">
        <title>Isopedia documentation - Xinchang Zheng</title>


        <!-- Custom HTML head -->

        <meta name="description" content="">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="theme-color" content="#ffffff">

        <link rel="shortcut icon" href="../favicon.png">
        <link rel="stylesheet" href="../css/variables.css">
        <link rel="stylesheet" href="../css/general.css">
        <link rel="stylesheet" href="../css/chrome.css">
        <link rel="stylesheet" href="../css/print.css" media="print">

        <!-- Fonts -->
        <link rel="stylesheet" href="../FontAwesome/css/font-awesome.css">
        <link rel="stylesheet" href="../fonts/fonts.css">

        <!-- Highlight.js Stylesheets -->
        <link rel="stylesheet" id="highlight-css" href="../highlight.css">
        <link rel="stylesheet" id="tomorrow-night-css" href="../tomorrow-night.css">
        <link rel="stylesheet" id="ayu-highlight-css" href="../ayu-highlight.css">

        <!-- Custom theme stylesheets -->
        <link rel="stylesheet" href="../theme/hide-nav.css">


        <!-- Provide site root and default themes to javascript -->
        <script>
            const path_to_root = "../";
            const default_light_theme = "light";
            const default_dark_theme = "navy";
            window.path_to_searchindex_js = "../searchindex.js";
        </script>
        <!-- Start loading toc.js asap -->
        <script src="../toc.js"></script>
    </head>
    <body>
    <div id="mdbook-help-container">
        <div id="mdbook-help-popup">
            <h2 class="mdbook-help-title">Keyboard shortcuts</h2>
            <div>
                <p>Press <kbd>←</kbd> or <kbd>→</kbd> to navigate between chapters</p>
                <p>Press <kbd>S</kbd> or <kbd>/</kbd> to search in the book</p>
                <p>Press <kbd>?</kbd> to show this help</p>
                <p>Press <kbd>Esc</kbd> to hide this help</p>
            </div>
        </div>
    </div>
    <div id="body-container">
        <!-- Work around some values being stored in localStorage wrapped in quotes -->
        <script>
            try {
                let theme = localStorage.getItem('mdbook-theme');
                let sidebar = localStorage.getItem('mdbook-sidebar');

                if (theme.startsWith('"') && theme.endsWith('"')) {
                    localStorage.setItem('mdbook-theme', theme.slice(1, theme.length - 1));
                }

                if (sidebar.startsWith('"') && sidebar.endsWith('"')) {
                    localStorage.setItem('mdbook-sidebar', sidebar.slice(1, sidebar.length - 1));
                }
            } catch (e) { }
        </script>

        <!-- Set the theme before any content is loaded, prevents flash -->
        <script>
            const default_theme = window.matchMedia("(prefers-color-scheme: dark)").matches ? default_dark_theme : default_light_theme;
            let theme;
            try { theme = localStorage.getItem('mdbook-theme'); } catch(e) { }
            if (theme === null || theme === undefined) { theme = default_theme; }
            const html = document.documentElement;
            html.classList.remove('light')
            html.classList.add(theme);
            html.classList.add("js");
        </script>

        <input type="checkbox" id="sidebar-toggle-anchor" class="hidden">

        <!-- Hide / unhide sidebar before it is displayed -->
        <script>
            let sidebar = null;
            const sidebar_toggle = document.getElementById("sidebar-toggle-anchor");
            if (document.body.clientWidth >= 1080) {
                try { sidebar = localStorage.getItem('mdbook-sidebar'); } catch(e) { }
                sidebar = sidebar || 'visible';
            } else {
                sidebar = 'hidden';
                sidebar_toggle.checked = false;
            }
            if (sidebar === 'visible') {
                sidebar_toggle.checked = true;
            } else {
                html.classList.remove('sidebar-visible');
            }
        </script>

        <nav id="sidebar" class="sidebar" aria-label="Table of contents">
            <!-- populated by js -->
            <mdbook-sidebar-scrollbox class="sidebar-scrollbox"></mdbook-sidebar-scrollbox>
            <noscript>
                <iframe class="sidebar-iframe-outer" src="../toc.html"></iframe>
            </noscript>
            <div id="sidebar-resize-handle" class="sidebar-resize-handle">
                <div class="sidebar-resize-indicator"></div>
            </div>
        </nav>

        <div id="page-wrapper" class="page-wrapper">

            <div class="page">
                <div id="menu-bar-hover-placeholder"></div>
                <div id="menu-bar" class="menu-bar sticky">
                    <div class="left-buttons">
                        <label id="sidebar-toggle" class="icon-button" for="sidebar-toggle-anchor" title="Toggle Table of Contents" aria-label="Toggle Table of Contents" aria-controls="sidebar">
                            <i class="fa fa-bars"></i>
                        </label>
                        <button id="theme-toggle" class="icon-button" type="button" title="Change theme" aria-label="Change theme" aria-haspopup="true" aria-expanded="false" aria-controls="theme-list">
                            <i class="fa fa-paint-brush"></i>
                        </button>
                        <ul id="theme-list" class="theme-popup" aria-label="Themes" role="menu">
                            <li role="none"><button role="menuitem" class="theme" id="default_theme">Auto</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="light">Light</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="rust">Rust</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="coal">Coal</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="navy">Navy</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="ayu">Ayu</button></li>
                        </ul>
                        <button id="search-toggle" class="icon-button" type="button" title="Search (`/`)" aria-label="Toggle Searchbar" aria-expanded="false" aria-keyshortcuts="/ s" aria-controls="searchbar">
                            <i class="fa fa-search"></i>
                        </button>
                    </div>

                    <h1 class="menu-title">Xinchang Zheng</h1>

                    <div class="right-buttons">
                        <a href="../print.html" title="Print this book" aria-label="Print this book">
                            <i id="print-button" class="fa fa-print"></i>
                        </a>

                    </div>
                </div>

                <div id="search-wrapper" class="hidden">
                    <form id="searchbar-outer" class="searchbar-outer">
                        <div class="search-wrapper">
                            <input type="search" id="searchbar" name="searchbar" placeholder="Search this book ..." aria-controls="searchresults-outer" aria-describedby="searchresults-header">
                            <div class="spinner-wrapper">
                                <i class="fa fa-spinner fa-spin"></i>
                            </div>
                        </div>
                    </form>
                    <div id="searchresults-outer" class="searchresults-outer hidden">
                        <div id="searchresults-header" class="searchresults-header"></div>
                        <ul id="searchresults">
                        </ul>
                    </div>
                </div>

                <!-- Apply ARIA attributes after the sidebar and the sidebar toggle button are added to the DOM -->
                <script>
                    document.getElementById('sidebar-toggle').setAttribute('aria-expanded', sidebar === 'visible');
                    document.getElementById('sidebar').setAttribute('aria-hidden', sidebar !== 'visible');
                    Array.from(document.querySelectorAll('#sidebar a')).forEach(function(link) {
                        link.setAttribute('tabIndex', sidebar === 'visible' ? 0 : -1);
                    });
                </script>

                <div id="content" class="content">
                    <main>
                        <h1 id="about-the-isopedia"><a class="header" href="#about-the-isopedia">About the Isopedia <img src="./img/logo3.png" align="right" alt="" width=120 /></a></h1>
<p><strong>Isopedia</strong> is a scalable tool designed for the analysis of hundreds to thousands of long-read transcriptome datasets simultaneously by read-level indexing approach. It provides two key capabilities:</p>
<p>Population-level transcript quantification and frequency profiling — With a single command, Isopedia can quantify isoform expression and estimate their occurrence frequencies across large cohorts using minimal computational resources.</p>
<p>Isoform diversity exploration and visualization — Isopedia enables systematic analysis of fusion genes and specific splicing events across populations, offering insights into transcript diversity beyond individual samples.</p>
<h1 id="table-of-content"><a class="header" href="#table-of-content">Table of Content</a></h1>
<ul>
<li><a href="#quick-start">Quick Start</a></li>
<li><a href="#how-it-works">How it works</a></li>
<li><a href="#download-pre-built-index">Download pre-built index</a></li>
<li><a href="#usage">Usage</a>
<ul>
<li><a href="#annotate-search-isoformstranscripts">Annotate (search) isoforms/transcripts</a></li>
<li><a href="#annotate-search-fusion-genes">Annotate (search) fusion genes</a></li>
<li><a href="#find-potential-fusion-genes">Find potential fusion genes</a></li>
</ul>
</li>
<li><a href="#installation">Installation</a></li>
<li><a href="#quick-qa">Quick Q&amp;A</a></li>
</ul>
<h1 id="quick-start"><a class="header" href="#quick-start">Quick Start</a></h1>
<p>Iospedida has two binaries: <code>isopedia</code> and <code>isopedia-tools</code>. the main binary <code>isopedia</code> is used for all the main functions, and <code>isopedia-tools</code> has some helper functions.</p>
<p><strong>Download prebuild index and run</strong></p>
<pre><code class="language-bash"># install isopedia from conda
conda install -y zhengxinchang::isopedia

# query transcripts
isopedia isoform -i index/ -g query.gtf -o out.isoform.tsv.gz

# query one fusion gene (two breakpoints)
isopedia fusion  -i index/ -p chr1:181130,chr1:201853853 -o out.fusion.tsv.gz

# query multiple fusion genes
isopedia fusion  -i index/ -P fusion_query.bed -o out.fusion.tsv.gz

# query gene regions and discover potential fusion events
isopedia fusion  -i index/ -g gene.gtf -o out.fusion.discovery.tsv.gz

# query a splice junction and visualize it
isopedia splice  -i index/ -s 17:7675236,17:7675993  -o out.splice.tsv.gz
python script/isopedia-splice-viz.py  -i out.splice.tsv.gz -g gencode.v47.basic.annotation.gtf  -t script/temp.html  -o isopedia-splice-view
</code></pre>
<p>For indexing GTF files, please refer to <a href="./doc/gtf.html">Indexing GTF Files</a> section.</p>
<h1 id="how-it-works"><a class="header" href="#how-it-works">How it works</a></h1>
<p><img src="./img/how-it-works.png" alt="how-it-works" /></p>
<p>The workflow of Isopedia involves several key steps, including isoform profiling, merging, indexing, and quering. Users can start by profiling isoform signals from individual BAM files, then merge the results to build a comprehensive index. Once the index is ready, it can be used to qeury isoforms, fusion genes, and explore splice junctions across multiple samples. Users can also visualize specific splicing events using the provided visualization tools.</p>
<p>Isopedia comes with pre-built indexes from hundreds of publicly available long-read RNA-seq datasets, which can be used directly for isoform and fusion gene annotation.</p>
<p><img src="./img/how-it-works2.png" alt="how-it-works2" /></p>
<p>This figure dipicts how Isopedia determines a positive hit for a query in different sinarios.</p>
<h1 id="download-pre-built-index"><a class="header" href="#download-pre-built-index">Download pre-built index</a></h1>
<p>[place holder for index download link]</p>
<h1 id="build-your-own-index"><a class="header" href="#build-your-own-index">Build your own index</a></h1>
<p>Isopedia supports building local index in your own datasets. prerequests are listed below:</p>
<ol>
<li>Latest isopedia binaries</li>
<li>A set of mapped bam files(sorted bam are not required)</li>
<li>A manifest file that describe the sample name, isoform file path, and other optional meta data in tabular(\t sperated and with a header line) format.</li>
</ol>
<p>Example manifest file:</p>
<pre><code>sample_name   path   platform
HG002_pb_chr22   /path/to/hg002_pb_chr22.isoform.gz   PacBio
HG002_ont_chr22   /path/to/hg002_ont_chr22.isoform.gz   ONT

</code></pre>
<h2 id="example-workflow"><a class="header" href="#example-workflow">Example workflow</a></h2>
<pre><code class="language-bash"># make sure isopedia in your $PATH or use absolute path to the binaries.

# download the toy_ex 
git clone https://github.com/zhengxinchang/isopedia &amp;&amp; cd isopedia/toy_ex/

# profile isoform signals on each bam individually
isopedia profile -b ./chr22.pb.grch38.bam -o ./hg002_pb_chr22.isoform.gz
isopedia profile -b ./chr22.ont.grch38.bam -o ./hg002_ont_chr22.isoform.gz

# make a manifest.tsv(tab-seprated) for *.isoform.gz files. example can be found at ./manifest.tsv

# merging, only first two column will be read in this step.
isopedia merge -i manifest.tsv -o index/

# build index. provide the same manifest file, the rest of meta columns will be read.
isopedia index  -i index/ -m manifest.tsv 

# test your index by run a small annotation task.
isopedia isoform -i index/ -g gencode.v47.basic.chr22.gtf -o out.isoform.tsv.gz

</code></pre>
<h1 id="usage"><a class="header" href="#usage">Usage</a></h1>
<h2 id="queryquantification-transcripts"><a class="header" href="#queryquantification-transcripts">Query(quantification) transcripts</a></h2>
<h3 id="purpose"><a class="header" href="#purpose">Purpose:</a></h3>
<p>search transcripts from input gtf file and return how many samples in the index have evidence.</p>
<h3 id="example"><a class="header" href="#example">Example:</a></h3>
<pre><code class="language-bash">isopedia isoform -i index/ -g query.gtf -o out.tsv.gz
</code></pre>
<p>key parameters:</p>
<p><code>--min-read(-m)</code> minimal support read in each sample to define a postive sample</p>
<p><code>--flank(-f)</code> flank base pairs when searching splice sites. large value will slow down the run time but allow more wobble splice site.</p>
<details>
<summary>
All parameters:
</summary>
<pre><code class="language-bash">Usage: isopedia isoform [OPTIONS] --idxdir &lt;IDXDIR&gt; --gtf &lt;GTF&gt; --output &lt;OUTPUT&gt;

Options:
  -i, --idxdir &lt;IDXDIR&gt;
          Path to the index directory

  -g, --gtf &lt;GTF&gt;
          Path to the GTF file

  -f, --flank &lt;FLANK&gt;
          Flanking size (in bases) before and after the position
          
          [default: 10]

  -m, --min-read &lt;MIN_READ&gt;
          Minimum number of reads required to define a positive sample
          
          [default: 1]

  -o, --output &lt;OUTPUT&gt;
          Output file for search results

  -w, --warmup-mem &lt;WARMUP_MEM&gt;
          Memory size to use for warming up (in gigabytes). Example: 4GB. Increasing this will significantly improve performance; set it as large as your system allows
          
          [default: 4]

  -c, --cached_nodes &lt;LRU_SIZE&gt;
          Maximum number of cached nodes per tree
          
          [default: 10000]

  -h, --help
          Print help (see a summary with '-h')

  -V, --version
          Print version

</code></pre>
</details>
<h3 id="output"><a class="header" href="#output">Output</a></h3>
<p>The output of the search command is a tab-separated file with the following columns:</p>
<div class="table-wrapper"><table><thead><tr><th>Column name</th><th>Description</th></tr></thead><tbody>
<tr><td>chrom</td><td>Chromosome</td></tr>
<tr><td>start</td><td>Start position of the query transcript</td></tr>
<tr><td>end</td><td>End position of the query transcript</td></tr>
<tr><td>length</td><td>Length of the query transcript</td></tr>
<tr><td>exon_count</td><td>Number of exons in the query transcript</td></tr>
<tr><td>trans_id</td><td>Transcript ID</td></tr>
<tr><td>gene_id</td><td>Gene ID</td></tr>
<tr><td>confidence</td><td>Confidence value for detecting the query transcript in the index</td></tr>
<tr><td>detected</td><td>Whether at least one sample supports this transcript with ≥ <code>--min-read</code> reads</td></tr>
<tr><td>min_read</td><td>Minimum number of reads to define a positive sample</td></tr>
<tr><td>positive_count/sample_size</td><td>Positive count / sample size</td></tr>
<tr><td>attributes</td><td>Original attributes of the transcript from the input GTF file</td></tr>
<tr><td>FORMAT</td><td>Format of the values in each sample column</td></tr>
<tr><td>sample1</td><td>Values</td></tr>
<tr><td>…</td><td>…</td></tr>
<tr><td>sampleN</td><td>Values</td></tr>
</tbody></table>
</div>
<p>There are a few columns can be used to filter the results.</p>
<p><code>detected</code> this binary value indicates if at least one sample has evidence to support the query transcirpt. it can be used to quickly filterout transcirpts without evidence.</p>
<p><code>positive_count/sample_size</code> this value is a combination of two values. it indicates how many samples have engouth evidence(defined by <code>--min-read</code>). it can be used to quckly filter the transcirpts that have at least several samples in the index.</p>
<p><code>confidence</code> a value that summarize the confidence of observing a transcript in the entire index</p>
<p><em>CPM</em> values are provided in each sample column, which is defined as:</p>
<p>$$CPM=\frac{ \text{Number of support reads for the query transcript}} {\text{Total number of valid reads in the sample}} * 1,000,000$$</p>
<details>
<p>$$C = \frac{k}{n}* (\prod_{i}^{n}CPM_{i})^{1/n} *G$$</p>
<p>where $n$ is the total number of samples in the index. $k$ is the sample number that found evidence(at least 1 support read) for a query. $CPM_{i}$ is the count per million value of the transcript in the sample $i$, which is defined as:</p>
<p>$$CPM_{i}=\frac{ \text{Number of support reads for the query transcript}} {\text{Total number of valid reads in the sample }i} * 1,000,000$$</p>
<p>$G$ is the GINI coefficient in positive samples$(i=0..k)$:</p>
<p>$$G = 2 \frac{\sum_{i=1}^{n} i*CPM_{i}}{n \sum_{i=1}^{n} CPM_{i} } - \frac{n+1}{n}$$</p>
</details>
<h2 id="query-fusion-gene-breakpoints"><a class="header" href="#query-fusion-gene-breakpoints">Query fusion gene breakpoints</a></h2>
<h3 id="purpose-1"><a class="header" href="#purpose-1">Purpose:</a></h3>
<p>This command is used to search for evidence of specific gene fusion events in the index based on provided breakpoints.</p>
<h3 id="example-1"><a class="header" href="#example-1">Example:</a></h3>
<pre><code class="language-bash"># query a single fusion
isopedia fusion -i index/ -f 10 -p chr1:pos1,chr2:pos2 -o fusion.anno.bed.gz

# query multiple fusions at the same time
isopedia fusion -i index/ -f 10 -P fusion_breakpoints.bed -o fusion_all.anno.bed.gz
</code></pre>
<p>key parameters:</p>
<p><code>--min-read(-m)</code> minimal support read in each sample to define a postive sample</p>
<p><code>--flank(-f)</code> flank base pairs when searching splice sites. large value will slow down the run time but allow more wobble splice site.</p>
<details>
<summary>
All parameters:
</summary>
<pre><code class="language-bash">Usage: isopedia fusion [OPTIONS] --idxdir &lt;IDXDIR&gt; --output &lt;OUTPUT&gt;

Options:
  -i, --idxdir &lt;IDXDIR&gt;
          index directory

  -p, --pos &lt;POS&gt;
          two breakpoints for gene fusion to be search(-p chr1:pos1,chr2:pos2)

  -P, --pos-bed &lt;POS_BED&gt;
          bed file that has the breakpoints for gene fusions. First four columns are chr1, pos1, chr2, pos2, and starts from the fifth column is the fusion id

  -G, --gene-gtf &lt;GENE_GTF&gt;
          bed file that has the start-end positions of the genes, used to find any possible gene fusions within the provided gene regions

  -f, --flank &lt;FLANK&gt;
          flank size for search, before and after the position
          
          [default: 10]

  -m, --min-read &lt;MIN_READ&gt;
          minimal reads to define a positive sample
          
          [default: 1]

  -o, --output &lt;OUTPUT&gt;
          output file for search results

      --debug
          debug mode

  -c, --cached_nodes &lt;LRU_SIZE&gt;
          number of cached nodes for each tree in maximal
          
          [default: 1000000]

  -h, --help
          Print help (see a summary with '-h')

  -V, --version
          Print version
</code></pre>
</details>
<h3 id="output-1"><a class="header" href="#output-1">Output</a></h3>
<div class="table-wrapper"><table><thead><tr><th>Column name</th><th>Description</th></tr></thead><tbody>
<tr><td>chr1</td><td>Chromosome of the first breakpoint</td></tr>
<tr><td>pos1</td><td>Position of the first breakpoint</td></tr>
<tr><td>chr2</td><td>Chromosome of the second breakpoint</td></tr>
<tr><td>pos2</td><td>Position of the second breakpoint</td></tr>
<tr><td>id</td><td>Event or transcript ID</td></tr>
<tr><td>min_read</td><td>Minimum number of reads required</td></tr>
<tr><td>sample_size</td><td>Total number of samples considered</td></tr>
<tr><td>positive_sample_count</td><td>Number of positive samples</td></tr>
<tr><td>sample1</td><td>Value/status in sample1</td></tr>
<tr><td>...</td><td>...</td></tr>
<tr><td>sampleN</td><td>Value/status in sampleN</td></tr>
</tbody></table>
</div>
<h2 id="find-potential-fusion-genes"><a class="header" href="#find-potential-fusion-genes">Find potential fusion genes</a></h2>
<h3 id="purpose-2"><a class="header" href="#purpose-2">Purpose:</a></h3>
<p>Query candidate fusion genes within specified gene regions. It identifies potential fusion events by examining all possible gene pairs within the provided regions and reporting those with supporting evidence in the index.</p>
<h3 id="example-2"><a class="header" href="#example-2">Example:</a></h3>
<pre><code class="language-bash">
isopedia fusion -i index/  -G gene.gtf -o fusion.discovery.out.gz
</code></pre>
<p>key parameters:</p>
<p><code>--gene-gtf(-G)</code> a gtf file that has gene records. the rest of feature will be ignored.</p>
<p><code>--min-read(-m)</code> minimal support read in each sample to define a postive sample</p>
<p><code>--flank(-f)</code> flank base pairs when searching splice sites. large value will slow down the run time but allow more wobble splice site.</p>
<h3 id="output-2"><a class="header" href="#output-2">Output</a></h3>
<div class="table-wrapper"><table><thead><tr><th>Column name</th><th>Description</th></tr></thead><tbody>
<tr><td>gene1_name</td><td>Name of gene 1</td></tr>
<tr><td>gene1_id</td><td>ID of gene 1</td></tr>
<tr><td>gene2_name</td><td>Name of gene 2</td></tr>
<tr><td>gene2_id</td><td>ID of gene 2</td></tr>
<tr><td>chr1</td><td>Chromosome of gene 1</td></tr>
<tr><td>start1</td><td>Consensus start position for gene 1 mapped region</td></tr>
<tr><td>end1</td><td>Consensus end position for gene 1 mapped region</td></tr>
<tr><td>chr2</td><td>Chromosome of gene 2</td></tr>
<tr><td>start2</td><td>Consensus start position for gene 2 mapped region</td></tr>
<tr><td>end2</td><td>Consensus end position for gene 2 mapped region</td></tr>
<tr><td>total_evidences</td><td>Total number of supporting evidences</td></tr>
<tr><td>total_samples</td><td>Total number of samples supporting the event</td></tr>
<tr><td>splice_junctions_count1</td><td>Number of splice junctions supporting gene 1</td></tr>
<tr><td>splice_junctions_count2</td><td>Number of splice junctions supporting gene 2</td></tr>
<tr><td>Sample1</td><td>Number of supporting reads in sample1</td></tr>
<tr><td>...</td><td>...</td></tr>
<tr><td>SampleN</td><td>Number of supporting reads in sampleN</td></tr>
</tbody></table>
</div>
<h2 id="query-splice-junctions-and-visualize-isoforms"><a class="header" href="#query-splice-junctions-and-visualize-isoforms">Query splice junctions and visualize isoforms</a></h2>
<h3 id="purpose-3"><a class="header" href="#purpose-3">Purpose:</a></h3>
<p>This command is designed for cases where you have a specific splice junction of interest and want to explore its isoform context in detail. It provides both tabular output and visualization.</p>
<h3 id="example-3"><a class="header" href="#example-3">Example:</a></h3>
<pre><code class="language-bash">isopedia splice  -i index/ -p chr22:41100500-41101500  -o splice.out.gz
python script/isopedia-splice-viz.py  -i splice.out.gz -g gencode.v47.basic.annotation.gtf  -t script/temp.html  -o isopedia-splice-view

</code></pre>
<p>key parameters(isopedia splice):</p>
<p><code>--min-read(-m)</code> minimal support read in each sample to define a postive sample</p>
<p><code>--flank(-f)</code> flank base pairs when searching splice sites. large value will slow down the run time but allow more wobble splice site.</p>
<p>key parameters(isopedia-splice-viz.py):</p>
<p><code>-g/--gtf</code> a gtf file that has gene annotations. it will be used to annotate the splice junction and isoforms.</p>
<p><code>-t/--temp-html</code> a template html file that will be used to generate the interactive vislization.</p>
<details>
<summary>
All parameters:
</summary>
<pre><code class="language-bash">
Usage: isopedia splice [OPTIONS] --idxdir &lt;IDXDIR&gt; --output &lt;OUTPUT&gt;

Options:
  -i, --idxdir &lt;IDXDIR&gt;
          Path to the index directory

  -s, --splice &lt;SPLICE&gt;
          Splice junction in 'chr1:pos1,chr2:pos2' format

  -S, --splice-bed &lt;SPLICE_BED&gt;
          Path to splice junction bed file

  -f, --flank &lt;FLANK&gt;
          Flanking size (in bases) before and after the position
          
          [default: 10]

  -m, --min-read &lt;MIN_READ&gt;
          Minimum number of reads required to define a positive sample
          
          [default: 1]

  -o, --output &lt;OUTPUT&gt;
          Output file for search results

  -w, --warmup-mem &lt;WARMUP_MEM&gt;
          Memory size to use for warming up (in gigabytes). Example: 4GB. Increasing this will significantly improve performance; Set it to 0(default) if you only have small query and want to skip warming up step
          
          [default: 0]

  -c, --cached_nodes &lt;LRU_SIZE&gt;
          Maximum number of cached nodes per tree
          
          [default: 100000]

  -h, --help
          Print help (see a summary with '-h')

  -V, --version
          Print version
</code></pre>
<pre><code class="language-bash">
usage: isopedia-splice-viz.py [-h] -i INPUT [-g GTF] [-t TEMPLATE] -o OUTPUT

Visualize isoforms from isopedia-anno-splice output

options:
  -h, --help            show this help message and exit
  -i INPUT, --input INPUT
                        Input file, the file is the output from isopedia-anno-splice with single query mode.
  -g GTF, --gtf GTF     Reference GTF file
  -t TEMPLATE, --template TEMPLATE
                        Templates HTML file
  -o OUTPUT, --output OUTPUT
                        Output file

</code></pre>
</details>
<h3 id="output-3"><a class="header" href="#output-3">Output</a></h3>
<p>The output is a gzip-compressed file containing detailed information about the splice junction and associated isoforms. Each isoform record includes:</p>
<div class="table-wrapper"><table><thead><tr><th>Field Name</th><th>Description</th></tr></thead><tbody>
<tr><td>#id</td><td>Isoform identifier</td></tr>
<tr><td>chr1, pos1</td><td>Chromosome and position of splice donor</td></tr>
<tr><td>chr2, pos2</td><td>Chromosome and position of splice acceptor</td></tr>
<tr><td>total_evidence</td><td>Total number of supporting reads</td></tr>
<tr><td>cpm</td><td>Normalized counts (CPM)</td></tr>
<tr><td>matched_sj_idx</td><td>Index of the matched splice junction</td></tr>
<tr><td>dist_to_matched_sj</td><td>Distance to the matched splice junction</td></tr>
<tr><td>n_exons</td><td>Number of exons in the isoform</td></tr>
<tr><td>start_pos_left</td><td>Leftmost starting position of isoform</td></tr>
<tr><td>start_pos_right</td><td>Rightmost starting position of isoform</td></tr>
<tr><td>end_pos_left</td><td>Leftmost ending position of isoform</td></tr>
<tr><td>end_pos_right</td><td>Rightmost ending position of isoform</td></tr>
<tr><td>splice_junctions</td><td>List of splice junctions in the isoform</td></tr>
<tr><td>format</td><td>Format of record</td></tr>
<tr><td>ENCSR***</td><td>Per-sample evidence (columns for each dataset, e.g., ENCODE accessions)</td></tr>
</tbody></table>
</div>
<p>Visualization output example:
https://zhengxinchang.github.io/isopedia/</p>
<p><img src="./img/isopedia-splice-view.png" alt="isopedia-splice-view" /></p>
<h1 id="memory-usage"><a class="header" href="#memory-usage">Memory Usage</a></h1>
<h2 id="encode-long-read-rna-seq-datasets107-samples"><a class="header" href="#encode-long-read-rna-seq-datasets107-samples">ENCODE long-read RNA-seq datasets(107 samples)</a></h2>
<div class="table-wrapper"><table><thead><tr><th>Step</th><th>Peak Memory Usage (GB)</th></tr></thead><tbody>
<tr><td>isopedia merge</td><td>7.12</td></tr>
<tr><td>isopedia-idx</td><td>3.84</td></tr>
<tr><td>isopedia isoform(158K transcripts from GENCODE)</td><td>15.82</td></tr>
</tbody></table>
</div>
<h1 id="installation"><a class="header" href="#installation">Installation</a></h1>
<h2 id="install-from-conda"><a class="header" href="#install-from-conda">Install from conda</a></h2>
<pre><code class="language-bash">conda install zhengxinchang::isopedia
</code></pre>
<h2 id="check-out-the-latest-release"><a class="header" href="#check-out-the-latest-release">Check out the latest release</a></h2>
<p>https://github.com/zhengxinchang/isopedia/releases</p>
<p>Note that the <code>isopedia-&lt;version&gt;.linux.tar.gz</code> was compliled in Amazon Linux 2 with GCC 7.3, Glibc 2.26, and Binutils 2.29.1. It should work in most of Linux distrubtion, however, if your Linux distribution can not run it, you can still try to use the <code>isopedia-&lt;version&gt;.musl.tar.gz</code>(statically linked with musl).</p>
<h2 id="from-source-code"><a class="header" href="#from-source-code">From source code</a></h2>
<p>Rust, cargo, and musl are required for building the project from source.</p>
<pre><code class="language-bash">git clone https://github.com/zhengxinchang/isopedia.git
cd isopedia
cargo build --release
cargo build --release --target x86_64-unknown-linux-musl
</code></pre>
<!-- 
# Quick Q&A:

**Q1: What gap does Isopedia aim to fill?**
A: Long-read RNA sequencing often reveals a large number of novel isoforms. However, evaluating whether these novel isoforms are biologically meaningful—or simply artifacts caused by RNA degradation or sequencing errors—is challenging. Understanding both the existence and population frequency of a novel isoform is essential for downstream analyses, yet current approaches are limited in scalability and robustness.

**Q2: How does Isopedia address this problem?**
A: Rather than relying on model-based classification or paired sequencing datasets from the same sample, Isopedia takes a different approach. It searches for supporting evidence of novel isoforms across large-scale long-read transcriptome datasets, leveraging population-level data to distinguish true biological isoforms from noise.

**Q3: How does Isopedia manage large datasets and provide a practical solution for isoform assessment?**
A: Isopedia introduces several innovations for scalable and efficient isoform evaluation. It uses a B+ tree–based data structure to rapidly index and compare isoform-related genomic positions. In addition, it employs a read-level signal extraction algorithm to build a compact yet informative index of transcriptome data. When assessing a query isoform, Isopedia integrates evidence from splicing junctions and alignment quality to ensure a robust evaluation. The entire tool is implemented in Rust, offering high performance and a user-friendly interface.

**Q4: What is the best use case for Isopedia?**
A: Isopedia is designed to be integrated into standard long-read transcriptome analysis pipelines. After isoforms are identified using tools like IsoQuant, FLAMES, TALON, or others, their output GTF files can be passed to Isopedia. It then evaluates the isoforms against a background index constructed from hundreds of publicly available long-read transcriptome datasets, providing a scalable and population-aware assessment.
 -->
<h1 id="contact"><a class="header" href="#contact">Contact</a></h1>
<ul>
<li>zhengxc93@gmail.com</li>
<li>fritz.sedlazeck@bcm.edu</li>
</ul>

                    </main>

                    <nav class="nav-wrapper" aria-label="Page navigation">
                        <!-- Mobile navigation buttons -->
                            <a rel="prev" href="../home.html" class="mobile-nav-chapters previous" title="Previous chapter" aria-label="Previous chapter" aria-keyshortcuts="Left">
                                <i class="fa fa-angle-left"></i>
                            </a>

                            <a rel="next prefetch" href="../isopedia/doc/gtf.html" class="mobile-nav-chapters next" title="Next chapter" aria-label="Next chapter" aria-keyshortcuts="Right">
                                <i class="fa fa-angle-right"></i>
                            </a>

                        <div style="clear: both"></div>
                    </nav>
                </div>
            </div>

            <nav class="nav-wide-wrapper" aria-label="Page navigation">
                    <a rel="prev" href="../home.html" class="nav-chapters previous" title="Previous chapter" aria-label="Previous chapter" aria-keyshortcuts="Left">
                        <i class="fa fa-angle-left"></i>
                    </a>

                    <a rel="next prefetch" href="../isopedia/doc/gtf.html" class="nav-chapters next" title="Next chapter" aria-label="Next chapter" aria-keyshortcuts="Right">
                        <i class="fa fa-angle-right"></i>
                    </a>
            </nav>

        </div>




        <script>
            window.playground_copyable = true;
        </script>


        <script src="../elasticlunr.min.js"></script>
        <script src="../mark.min.js"></script>
        <script src="../searcher.js"></script>

        <script src="../clipboard.min.js"></script>
        <script src="../highlight.js"></script>
        <script src="../book.js"></script>

        <!-- Custom JS scripts -->



    </div>
    </body>
</html>
